### PR TITLE
boot_nic_with_iommu: Refactor code to support other architectures

### PR DESCRIPTION
**Pull request by PaulYuuu**
Refactor boot_nic_with_intel_iommu to support run it on other
architectures.

ID: 1879443
Signed-off-by: Yihuang Yu <yihyu@redhat.com>

### CODE REVIEW

**Comment by PaulYuuu** [comment]
Hello @yanglei-rh, please help to review, thanks.

**Comment by yanglei-rh** [comment]
Hi @PaulYuuu , it can be test pass, But this is a Intel_cpu  polarion test case. According to the current code,it can be tested on AMD, is this the result we expect?

**Comment by PaulYuuu** [comment]
> Hi @PaulYuuu , it can be test pass, But this is a Intel_cpu polarion test case. According to the current code,it can be tested on AMD, is this the result we expect?

@yanglei-rh I think `iommu_platform` can be tested on all platforms. On AMD, we can remove `intel-iommu` device in the qemu command line. If there has another issue(e.g: add amd_iommu in the kernel line) block the test, I can remove AMD support this time.

**Comment by yanglei-rh** [comment]
> > Hi @PaulYuuu , it can be test pass, But this is a Intel_cpu polarion test case. According to the current code,it can be tested on AMD, is this the result we expect?
> 
> @yanglei-rh I think `iommu_platform` can be tested on all platforms. On AMD, we can remove `intel-iommu` device in the qemu command line. If there has another issue(e.g: add amd_iommu in the kernel line) block the test, I can remove AMD support this time.

Ok, I get it. it has test passed on the AMD platform, and qemu command line meets the requirements. I will add test results later.

**Comment by yanglei-rh** [comment]
LGTM
Acked-by: Lei Yang

Test Pass:
AMD Platform:
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.q35: PASS (49.48 s)
(1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.q35: PASS (120.12 s)
Intel Platform:
(3/3) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.q35: PASS (162.89 s)
(2/2) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.boot_nic_with_iommu.q35: PASS (139.42 s)

**Comment by yanan-fu** [comment]
There is a new mechanism to distinguish different platform, you can try  `only HostCpuVendor.intel`

**Comment by PaulYuuu** [comment]
> There is a new mechanism to distinguish different platform, you can try `only HostCpuVendor.intel`

No, this patch is in order to support this case on all platforms.

**Comment by yanan-fu** [comment]
> > There is a new mechanism to distinguish different platform, you can try `only HostCpuVendor.intel`
> 
> No, this patch is in order to support this case on all platforms.

AH~  my careless,  del/add, ignore me~

File: qemu/tests/cfg/boot_nic_with_iommu.cfg [hunk 1]
```diff
@@ -0,0 +1,17 @@
+- boot_nic_with_iommu:
+    x86_64, i386:
+        only q35
+        no WinXP WinVista Win7 Win8 Win8.1 Win2003
+        no Win2008 Win2008..r2 Win2012 Win2012..r2
+        machine_type_extra_params = "kernel-irqchip=split"
+        HostCpuVendor.intel:
+            intel_iommu = yes
+        Linux:
+            enable_guest_iommu = yes
```
Comment (yanan-fu) [inline qemu/tests/cfg/boot_nic_with_iommu.cfg:10]: What will it happen for amd ? No intel-iommu device, but enable guest iommu ? Is this the expected ?

**Comment by PaulYuuu** [comment]
Test passed on PPC:
```
 (1/1) Host_RHEL.m8.u3.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.ppc64le.io-github-autotest-qemu.boot_nic_with_iommu: PASS (134.75 s)
```
Thanks all.

File: qemu/tests/cfg/boot_nic_with_iommu.cfg (hunk 1 above)
Comment (PaulYuuu) [reply qemu/tests/cfg/boot_nic_with_iommu.cfg:10]: Left this comment. On AMD, only enable iommu_platform for all virtio devices. `enable_guest_iommu` will check the CPU vendor at first and then add a parameter to the kernel command line.

Comment (yanan-fu) [reply qemu/tests/cfg/boot_nic_with_iommu.cfg:10]: Better  'enable_guest_iommu = yes'  with 'HostCpuVendor.intel'  only,   But anyway, patch already be merged and result is same, leave it ~

**Pull request merged**